src/nixpacks/builder/docker/dockerfile_generation.rs

```diff
@@ -148,7 +148,7 @@ impl DockerfileGenerator for BuildPlan {
                 // Pull the variables in from docker `--build-arg`
                 variables
                     .iter()
-                    .map(|var| var.0.to_string())
+                    .map(|var| format!("{}=\"{}\"", var.0.trim(), var.1.trim()))
```

Comment:
This line changes the Dockerfile output from:

```Dockerfile
ARG ELIXIR_ERL_OPTIONS LANG LANGUAGE MIX_ENV NIXPACKS_METADATA
```

To:

```Dockerfile
ARG ELIXIR_ERL_OPTIONS="+fnu" LANG="en_US.UTF-8" LANGUAGE="en_US:en" MIX_ENV="prod" NIXPACKS_METADATA="elixir"
```

Looking at the [docker arg docs](https://docs.docker.com/reference/dockerfile/#arg) and from my own testing, this simply sets a default value for the ARG but it can be overridden by setting a variable.

Reply:
without this line, the rest of the dockerfile doesn't actually inherit values like MIX_ENV for the build phases and then we need to find a different workaround (like prefixing commands with `MIX_ENV=prod mix local.rebar --force`)

Reply:
I believe we have intentionally not done this because then all variables will be logged in the build logs. The current way of injecting build variables prevents that.

Reply:
but this is injecting default values of environment variables. isn't it okay for these ones to be logged? like `MIX_ENV= prod`

there shouldn't be any user specific secrets set as defaults afaik.

Reply:
@coffee-cup is this the only discussion point that's blocking? I will comb through the codebase a bit and see if there are any defaults that we might handle differently. the alternative here is to add some flag to explicitly say "in this case, please fill in the defaults"